src/MongoDB.Driver.Core/MongoException.cs

```diff
@@ -57,6 +62,36 @@ public MongoException(string message, Exception innerException)
         public MongoException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
+            _errorLabels = (List<string>)info.GetValue(nameof(_errorLabels), typeof(List<String>));
+        }
+#endif
+
+        // public properties
+        /// <summary>
+        /// Gets the error labels.
+        /// </summary>
+        /// <value>
+        /// The error labels.
+        /// </value>
+        public IReadOnlyList<string> ErrorLabels => _errorLabels;
+
+        // public methods
+        /// <summary>
+        /// Adds an error label.
+        /// </summary>
+        /// <param name="errorLabel">The error label.</param>
+        public void AddErrorLabel(string errorLabel)
+        {
+            Ensure.IsNotNull(errorLabel, nameof(errorLabel));
+            _errorLabels.Add(errorLabel);
+        }
+
+#if NET45
+        /// <inheritdoc/>
+        public override void GetObjectData(SerializationInfo info, StreamingContext context)
```

Comment:
Can you explain what this is for?

Reply:
This is required because MongoException is [Serializable].

It is new because before MongoException had no fields. But it was already [Serializable].